### PR TITLE
Testing fixes

### PR DESCRIPTION
**Pull request by djmitche**
This is a bunch of minor refactors to use newer `taskcluster-lib-*`, with the end result that it's a lot easier and faster to run the tests.

### CODE REVIEW

File: bin/server.js [hunk 1]
```diff
@@ -10,189 +10,210 @@ var exchanges     = require('../auth/exchanges');
...
-      validator = await base.validator({
+var loader        = require('taskcluster-lib-loader');
+var app           = require('taskcluster-lib-app');
+
+// Create component loader
+let load = loader({
+  cfg: {
+    requires: ['profile'],
+    setup: ({profile}) => {
+      return base.config({
```
Comment (jonasfj) [inline bin/server.js]: I know it's a pain, but upgrading base and using new config is also worth it...

Lot's of good things have been happening in API, from query string support for things like continuation tokens... to things like better error handlings and error messages...

File: auth/data.js [hunk 2]
```diff
@@ -22,7 +22,7 @@ var Client = base.Entity.configure({
      * - lastRotated      // Last time accessToken was reset
      * (more properties may be added in the future)
      */
-    details:        base.Entity.types.JSON
+    details:        Entity.types.JSON
   },
   context:          ['resolver']
```
Comment (jonasfj) [inline auth/data.js]: We could probably get rid of this circlar dependency... and take it as a parameter for `.json`

File: bin/server.js (hunk 1 above)
Comment (djmitche) [inline bin/server.js]: Sure, but it's not necessary here.  Well, in a subsequent patch, query string support might be useful (for listClients), but not in this PR.

File: auth/data.js (hunk 2 above)
Comment (jonasfj) [inline auth/data.js]: with `loader` it's pretty obvious that circular dependencies are bad.

**Comment by jonasfj** [comment]
`r+`

My comment on fixing the circular dependency shouldn't block anything... it's just a nice to have.. .that'll probably make us happier further down the line... (also it would simplify tests)

**Pull request merged**